vorc_gazebo/src/gymkhana_scoring_plugin.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "vorc_gazebo/gymkhana_scoring_plugin.hh"
+
+/////////////////////////////////////////////////
+GymkhanaScoringPlugin::GymkhanaScoringPlugin()
+{
+  gzmsg << "GymkhanaScoringPlugin loaded" << std::endl;
+}
+
+/////////////////////////////////////////////////
+GymkhanaScoringPlugin::~GymkhanaScoringPlugin()
+{
+}
+
+/////////////////////////////////////////////////
+void GymkhanaScoringPlugin::Load(gazebo::physics::WorldPtr _world,
+  sdf::ElementPtr _sdf)
+{
+  // Base class, also binds the update method for the base class
+  ScoringPlugin::Load(_world, _sdf);
+
+  this->updateConnection = gazebo::event::Events::ConnectWorldUpdateBegin(
+    std::bind(&GymkhanaScoringPlugin::Update, this));
+
+  // Setup ROS node and publisher
+  this->rosNode.reset(new ros::NodeHandle());
+
+  this->channelSub = this->rosNode->subscribe(
+    "/vorc/gymkhana_channel/task/info", 5,
+    &GymkhanaScoringPlugin::ChannelCallback, this);
+
+  this->blackboxSub = this->rosNode->subscribe(
+    "/vorc/gymkhana_blackbox/task/info", 5,
+    &GymkhanaScoringPlugin::BlackboxCallback, this);
+}
+
+/////////////////////////////////////////////////
+void GymkhanaScoringPlugin::Update()
+{
+  // Nothing to do if the task is not in "running" state.
+  if (this->ScoringPlugin::TaskState() != "running")
+    return;
+
+  // Check channel navigation is finished
+  // If not, invalidate black box station-keeping score
```

Comment:
Does this comment apply?

Reply:
Yes, it's just 0 if channel navigation isn't finished.